components/cam/src/physics/cam/phys_grid.F90

```diff
@@ -2433,52 +2437,107 @@ subroutine print_cost_p()
    integer  :: unitn                  ! file unit number
    integer  :: signal                 ! handshake variable
    integer  :: lcid                   ! local chunk id
+   integer  :: tid                    ! thread id
    integer  :: cid                    ! global chunk id
    integer  :: owner                  ! chunk owner
    integer  :: ncols                  ! number of columns in chunk
    integer  :: ierr                   ! error return
 
-   real(r8) :: cost_lsum, cost_gsum   ! local and global sums of 
-                                      !  chunk costs
-   real(r8) :: avg_cost               ! average chunk cost
-   real(r8) :: avg_estcost            ! average estimated chunk cost
-   real(r8) :: cost                   ! chunk cost
-   real(r8) :: norm_cost              ! normalized chunk cost
-   real(r8) :: norm_est_cost          ! normalized estimated chunk cost
+   real(r8) :: lsums(4)               ! working arrays for local sums
+   real(r8) :: gsums(4)               ! working arrays for global sums
+   real(r8) :: ccest_lsum, ccest_gsum ! local and global sums of 
+                                      !  estimated normalized chunk costs
+   real(r8) :: ccost_lsum, ccost_gsum ! local and global sums of 
+                                      !  measured chunk costs
+   real(r8) :: tcest(0:nlthreads-1)   ! estimated normalized thread cost
+                                      !  for chunk loops in bc_physics 
+                                      !  and ac_physics
+   real(r8) :: pcest, pcest_sum       ! estimated normalized process cost
```

Comment:
Similar to another suggestion, I would personally prefer more verbose variable names like "proc_cost_est" instead of "pcest", but given the nice descriptions in the variable declarations it's not hard to figure things out, so it's not critical.

Reply:
I'll revisit these names as well, and hopefully come up with something more self-documenting. Thanks.

Reply:
@whannah1 , done. please take a look. I also made similar changes in create_chunks, pheap_adjust (now proc_heap_adjust) and cheap_adjust (now chnk_heap_adjust).